File: docs/static/examples/src/mouse-events/java/MouseEvents.java
import io.github.bernhardfritz.p8g.Sketch;

public class MouseEvents extends Sketch {

    private boolean once;

    @Override
    protected void draw() {
        if (!once) {
            background(220);
            once = true;
        }
    }

    @Override
    protected void mouseMoved() {
        ellipseMode(CENTER);
        ellipse(mouseX, mouseY, 50, 50);
    }

    @Override
    protected void mousePressed() {
        background(220);
    }

    public static void main(String[] args) {
        Sketch sketch = new MouseEvents();
        sketch.run(320, 320, "Mouse events");
    }
}

File: docs/static/examples/src/mouse-events/js/mouse-events.js
import p8g, {
  background,
  createCanvas,
  CENTER,
  ellipseMode,
  ellipse,
  mouseX,
  mouseY,
} from 'p8g.js';

let once = false;

p8g.draw = () => {
  if (!once) {
    background(220);
    once = true;
  }
};

p8g.mouseMoved = () => {
  ellipseMode(CENTER);
  ellipse(mouseX, mouseY, 50, 50);
};

createCanvas(320, 320);
